afs/models.py

```diff
@@ -155,35 +261,120 @@ def switch_repo(self, repo_name=None):
         :param str repo_name: (optional)The name of model repository.
         :return: None, repo_id, exception
         """
-        params = dict(name=repo_name)
-        resp = self._get(params=params)
-        if len(resp.json()) == 0:
-            return None
-        elif len(resp.json()) == 1:
-            self.repo_id = resp.json()[0]['uuid']
-            return self.repo_id
+        if self.api_version == 'v1':
+            params = dict(name=repo_name)
+            resp = self._get(params=params)
+            if len(resp.json()) == 0:
+                return None
+            elif len(resp.json()) == 1:
+                self.repo_id = resp.json()[0]['uuid']
+                return self.repo_id
+            else:
+                raise ValueError('There are multi model repositories from server response')
+                return None
         else:
-            raise ValueError('There are multi model repositories from server response')
-
+            raise NotImplementedError('v2 API is not support this method.')
+            return None
 
-    def get_latest_model_info(self, repo_name=None):
+    def get_latest_model_info(self, model_repository_name=None):
         """
         Get the latest model info, including created_at, tags, evaluation_result. (Support v2 API)
         
-        :param repo_name: (optional)The name of model repository.
+        :param model_repository_name: (optional)The name of model repository.
         :return: dict. the latest of model info in model repository.
         """
-        if repo_name:
-            self.switch_repo(repo_name)
-
         if self.api_version == 'v1':
+            if model_repository_name:
+                self.switch_repo(model_repository_name)
             resp = self._get(extra_paths=[self.repo_id, 'info'])
+            return resp.json()
         elif self.api_version == 'v2':
-            resp = self._get(extra_paths=[self.repo_id, 'last'])
+            model_id = self.get_model_id(model_repository_name=model_repository_name, last_one=True)
+
+            if model_id:
+                extra_paths = [self.repo_id, self.sub_entity_uri, model_id]
+                resp = self._get(extra_paths=extra_paths)
+                resource = resp.json()['resources'][0]
+                return resource
+            else:
+                raise ValueError('Model not found.')
+                return None
+
+    def get_model_info(self, model_name, model_repository_name=None):
```

Comment:
I think `get_model_info()` and `get_latest_model_info()` is duplicated. maybe use argument to control would be better.

Reply:
`get_latest_model_info()` is v1 method. In order to user can maintain their code in v1, I want to keep this method. `get_model_info ` is support in v2 method.

Reply:
If so, I think to raise a deprecated warning when using `get_latest_model_info()` and remove `v2` support from this function(or `NotImplementedError`). In `v2`, there will only `get_model_info()`.